packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -172,6 +187,133 @@ export function pointReducer(
+
+          if (!Layers.isPointsLayer(layer) || !pointList || !boundingRect)
+            return;
+
+          // Update all points by first transforming to the canvas's coordinate system
+          layer.points
+            .filter((_, index) => pointList.includes(index))
+            .forEach((curvePoint) => {
+              const decodedPoint = decodeCurvePoint(curvePoint, boundingRect);
+              if (!controlPointType) return state;
+
+              //TODO: Fix Asymmetric point values
+
+              if (
+                (controlPointType === 'curveFrom' ||
+                  controlPointType === 'curveTo') &&
+                (curveMode === 2 || curveMode === 3)
+              ) {
+                const oppositeControlPoint =
+                  controlPointType === 'curveFrom' ? 'curveTo' : 'curveFrom';
+
+                const delta = decodedPoint[controlPointType][axis] - amount;
+
+                const testing =
+                  mode === 'replace'
+                    ? decodedPoint[oppositeControlPoint][axis] + delta
+                    : decodedPoint[oppositeControlPoint][axis] + amount * -1;
+
+                decodedPoint[oppositeControlPoint] = {
+                  ...decodedPoint[oppositeControlPoint],
+                  [axis]: testing,
+                };
+
+                const testing2 =
+                  mode === 'replace'
+                    ? amount
+                    : decodedPoint[controlPointType][axis] + amount;
+
+                decodedPoint[controlPointType] = {
+                  ...decodedPoint[controlPointType],
+                  [axis]: testing2,
+                };
+              } else {
+                const newValue =
+                  mode === 'replace'
+                    ? amount
+                    : decodedPoint[controlPointType][axis] + amount;
+
+                decodedPoint[controlPointType] = {
+                  ...decodedPoint[controlPointType],
+                  [axis]: newValue,
+                };
+              }
+
+              const encodedPoint = encodeCurvePoint(decodedPoint, boundingRect);
+
+              //TODO: why are the controlPointTypes switched?
+              //if curveMode === 2 or 3 update the adjacent control point
+              if (controlPointType === 'curveFrom') {
+                curvePoint.curveTo = encodedPoint.curveTo;
+                if (curveMode === 2 || curveMode === 3) {
+                  curvePoint.curveFrom = encodedPoint.curveFrom;
+                }
+              } else {
+                curvePoint.curveFrom = encodedPoint.curveFrom;
+                if (curveMode === 2 || curveMode === 3) {
+                  curvePoint.curveTo = encodedPoint.curveTo;
+                }
+              }
+            });
+
+          const decodedPoints = layer.points.map((curvePoint) =>
+            decodeCurvePoint(curvePoint, boundingRect),
+          );
+
+          const layerPath = path(
```

Comment:
Maybe clearer as `const [minX, minY, maxX, maxY] = path(...)`, if that's accurate